projects/plugins/social/src/js/components/header/styles.module.scss

```diff
@@ -54,3 +54,8 @@
 		margin-top: calc( var( --spacing-base ) * 3 );
 	}
 }
+
+// Defined twice to avoid specificity issues.
```

Comment:
With `CSS Modules` we don't need to care about specificity, it hashes the class name into a unique value.

https://github.com/css-modules/css-modules#css-modules

Reply:
Sort of. The classes all get a unique value, but all CSS specificity order rules still apply. With the way the components are set up, the CSS for the component gets loaded after the custom CSS for the class I added in the Social admin page, which means the margin gets overruled:

<img width="327" alt="Screenshot 2022-09-09 at 12 09 54@2x" src="https://user-images.githubusercontent.com/1713699/189326873-b65bab51-13a6-4b4c-8198-c8ad8e9ac5a6.png">

By adding the class twice, we're [increasing the specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#increasing_specificity_by_duplicating_selector) while avoiding `!important`:

<img width="322" alt="image" src="https://user-images.githubusercontent.com/1713699/189327085-2dd93a2e-f164-49e7-8883-d80148240118.png">

Reply:
Ok. I got it, I thought you were doing it for other reasons, not overriding other styles, maybe it is something to include in the comment.

Reply:
That might work, but I feel like it's a bigger problem that needs to be solved on the `@jetpack/components` level. It seems like extra classes that get added to the existing components should **always** have priority over the components classes.

I'm not sure how to fix that—I haven't looked into it properly yet.

Reply:
> The component styles should be the priority at all moments

Is that true for all styles? Styles that can conflict with the layout, like we have here, will often need to be overridden. This solution is a hack that MDN says should be used sparingly if at all.

Do you think we should be opinionated about some styles rather than all?

Reply:
> Is that true for all styles?

IMO, it should be, our current approach for components is to make them consistent and customizable, if we need to override some of the default styles, it is a signal for one of the two things:

1) Designers moved out of our specifications, it happens sometimes and normally we go for them pointing it, to check if it was a mistake or if they want to update the component for that new way. Ex: p1661354188482839-slack-C03TA48NSUX 

2) We should open the component (via props) to allow end users to set that value. This is exactly what we did with `Text`, allowing user to set margin via props, here: https://github.com/Automattic/jetpack/pull/23682

Reply:
@danielpost Btw, looking at the code again, we could move out of this. The `margin: 0` implemented at the `.cut` class was used because it unset the default value for the `button` tag. Since we move to `div`, it could be removed.

Reply:
That's great that the margin can be removed.

> We should open the component (via props) to allow end users to set that value. This is exactly what we did with Text, allowing user to set margin via props

Mmmm, I'm not sure I agree with that, but I'd need to think about it some more! It's essentially reimplementing some CSS in JS, which is fine, but we could allow certain propeties to be overridden in CSS by being careful with specificity. Again the `Text` component needed to allow people to set box model properties, which is the same issue as here. I agree that we shouldn't open up all the CSS so things can be freely tweaked, but these kind of layout properties will often need to be set based on the context the component is being used. Adding props to all the components doesn't seem like the right solution if we can use CSS.

Having said all of that, we can remove the problem by not setting the marign on the component, so this becomes a discussion for another day 😄

Reply:
>  It's essentially reimplementing some CSS in JS

Not exactly like that. React components rely on top of one direction data flow, that's the props. We just use them to allow the user to control the component. It is the same as we do in other situations (disable/enable, etc..), but related to spacing. It already prevents you from creating a new whole CSS file to customize the component and ensure that spacings are following our specs, which is [`8px` grid](https://uxplanet.org/everything-you-should-know-about-8-point-grid-system-in-ux-design-b69cb945b18d) layout.

Reply:
This is an interesting discussion, and something I'd like to think about some more, but not on this PR 😄

The mixture of props CSS and classes is probably a necessary evil. It's good that we've got CSS classes defined for the grid.

Reply:
Sure. Agreed. Feel free to jump our channel or P2